tool/loadbalancer/manager.go

```diff
@@ -0,0 +1,120 @@
+package loadbalancer
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	fileutils "github.com/gravitational/planet/lib/utils/file"
+	"github.com/gravitational/planet/lib/utils/systemd"
+	"github.com/gravitational/trace"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// NewManager creates a new manager for loadbalancer
+func NewManager() *Manager {
+	storageCh := make(chan string, 5)
+	return &Manager{
+		storage:   NewStorage(storageCh),
+		storageCh: storageCh,
+	}
+}
+
+// Manager manages the haproxy setting and the start of the service
+type Manager struct {
+	storage   *Storage
+	storageCh <-chan string
+}
+
+// GetStorage return storage object
+func (m *Manager) GetStorage() *Storage {
+	return m.storage
+}
+
+// Run runs manager
+func (m *Manager) Run() {
+	ticker := time.NewTicker(time.Second * 30)
+	for {
+		select {
+		case <-ticker.C:
+			if err := m.reconfigure(); err != nil {
+				log.Warn("unable to reconfigure HAProxy service", err)
+			}
+		case <-m.storageCh:
+			if err := m.reconfigure(); err != nil {
+				log.Warn("unable to reconfigure HAProxy service", err)
+			}
+		}
+	}
+}
+
+// StopService disables and stops the service
+func (m *Manager) StopService() error {
+	err := systemd.DisableService(context.Background(), "haproxy.service")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (m *Manager) ensureService(needRestart bool) error {
+	service := "haproxy.service"
+	status, err := systemd.GetServiceStatus(service)
+	if err != nil {
+		return trace.Wrap(err, "Unable to get service status %s", service)
+	}
+	log.WithFields(log.Fields{
+		"service": service,
+		"status":  status,
+	}).Info("service info")
+	if status == "active" && needRestart {
+		err := systemd.Systemctl(context.Background(), "restart", service)
+		if err != nil {
+			return trace.Wrap(err, "Unable to restart service %s", service)
```

Comment:
```suggestion
			return trace.Wrap(err, "unable to restart service %s", service)
```
error messages need not be capitalized as they need to be composable.